### ISSUE TITLE
Not returning `null` for non-github URLs

### ISSUE THREAD

**Issue by revin** [zeke/github-url-to-object#19]
Hi all; looks like 2.2 introduced a breaking change where passing a non-github URL doesn't result in a `null` result like 2.1 did:

With 2.1:
```javascript
var gh = require('github-url-to-object')
console.log(gh('https://bitbucket.com/mark/bitbucketberg'))
// null
```

With 2.2:
```javascript
var gh = require('github-url-to-object')
console.log(gh('https://bitbucket.com/mark/bitbucketberg'))
/*
{ user: 'mark',
  repo: 'bitbucketberg',
  host: 'bitbucket.com',
  branch: 'master',
  apiHost: 'bitbucket.com/api/v3',
  tarball_url: 'https://bitbucket.com/api/v3/repos/mark/bitbucketberg/tarball/master',
  clone_url: 'https://bitbucket.com/mark/bitbucketberg',
  https_url: 'https://bitbucket.com/mark/bitbucketberg',
  travis_url: 'https://travis-ci.org/mark/bitbucketberg',
  zip_url: 'https://bitbucket.com/mark/bitbucketberg/archive/master.zip',
  api_url: 'https://bitbucket.com/api/v3/repos/mark/bitbucketberg' }
*/
```

I'm guessing it was an accident, since the README still says "If you provide a non-GitHub URL or a falsey value, you'll get null."

**Comment by revin** [comment]
Some more poking around revealed that it seems to depend on the structure of the URL:

```javascript
gh('https://example.com/') // null
gh('https://example.com/path') // null
gh('https://example.com/path/path')
/*
{ user: 'path',
  repo: 'path',
  host: 'example.com',
  branch: 'master',
  apiHost: 'example.com/api/v3',
  tarball_url: 'https://example.com/api/v3/repos/path/path/tarball/master',
  clone_url: 'https://example.com/path/path',
  https_url: 'https://example.com/path/path',
  travis_url: 'https://travis-ci.org/path/path',
  zip_url: 'https://example.com/path/path/archive/master.zip',
  api_url: 'https://example.com/api/v3/repos/path/path' }
*/
```
It seems that once the path part of my URL has enough slashes, I don't get `null` like I expect.

**Comment by zeke** [comment]
Pinging @appleton and @dmathieu, who I think accidentally introduced this bug when adding GitHub enterprise support. How might we proceed while continuing to support the expected `null` result as well as GitHub enterprise URLs?

**Comment by revin** [comment]
Thanks @zeke, really appreciate it :+1:

**Comment by revin** [comment]
Hey @zeke / @appleton / @dmathieu, just checking in; any progress or ideas here? Thanks! 😎

**Comment by zeke** [comment]
@revin sorry for letting this languish. I'll take a look today.

**Comment by zeke** [comment]
Hi @revin. I just released 2.2.3 which restores the old behavior of returning `null` if the given URL's hostname is not `github.com`. 

I've introduced an option for enterprise URLs:

```js
gh('git@ghe.example.com:heroku/heroku-flags.git', {enterprise: true})
```

This is a tricky case for knowing what semver to use, so @appleton and @dmathieu I'm sorry if this breaks your GHE stuff for a minute! I figured it best to release this a bug fix to continue supporting the pre-existing `null` behavior.
